src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -117,13 +121,48 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter);
         }
     };
 }
 
 using namespace RSS;
 
+QString computeEpisodeName(const QString &article)
+{
+    static const QRegularExpression episodeRegex("(?:^|[^A-Za-z0-9])(?:"
+
+                                                 //Format 1: s01e01
+                                                 "(?:s(\\d{1,})e(\\d{1,}))|"
+
+                                                 //Format 2: 01x01
+                                                 "(?:(\\d{1,})x(\\d{1,}))|"
+
+                                                 //Format 3: 2017.01.01
+                                                 "(\\d{4}\\.\\d{2}\\.\\d{2})"
+
+                                                 ")(?:[^A-Za-z0-9]|$)",
+                                                 QRegularExpression::CaseInsensitiveOption |
+                                                 QRegularExpression::ExtendedPatternSyntaxOption);
+
+    QRegularExpressionMatch match = episodeRegex.match(article);
+
+    // See if we can extract an season/episode number or date from the title
+    if (!match.hasMatch()) {
```

Comment:
Although not mandatory, but it is better to omit the curly braces if there is only one line (exactly, one statement) inside.

UPD: this comment affects some other places in your code too.